carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/Screenshot.java

```diff
@@ -379,22 +383,22 @@ private static void uploadToAmazonS3(File screenshot) {
             LOGGER.debug("there is no sense to continue as saving screenshots onto S3 is disabled.");
             return;
         }
-        String correlationId = UUID.randomUUID().toString();
-        TestType test = ZafiraListener.getTestbythread().get(Thread.currentThread().getId());
+        final String correlationId = UUID.randomUUID().toString();
+        final String ciTestId = ZafiraListener.getThreadCiTestId();
         try {
             ZafiraMessager.<MetaInfoMessage>custom(MetaInfoLevel.META_INFO, new MetaInfoMessage()
                     .addHeader("AMAZON_PATH", null)
                     .addHeader("AMAZON_PATH_CORRELATION_ID", correlationId));
             executorService.execute(() -> {
                 try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://52.53.207.202/sonarqube/coding_rules#rule_key=squid%3AS1141)